### PR TITLE
update version : hepmc

### PR DESCRIPTION
**Pull request by darmac**

### CODE REVIEW

File: var/spack/repos/builtin/packages/hepmc/package.py [hunk 1]
```diff
@@ -14,6 +14,10 @@ class Hepmc(CMakePackage):
     homepage = "http://hepmc.web.cern.ch/hepmc/"
     url      = "http://hepmc.web.cern.ch/hepmc/releases/hepmc2.06.09.tgz"
 
+    version('3.2.0', sha256='b59f301e2660db3c81e4fd4aa0701cc010463691ac57424edf4301e16a5944b4')
+    version('3.1.2', sha256='7df91b862cead2657ee54ab043ecf0a8f46da354c8a7ea90d8db2c435904bce0')
+    version('3.1.1', sha256='46098a7e9717921e436c265836c914d515cb831dde898f3f1d129468df7982e7')
+    version('3.1.0', sha256='ac02977d6bde18398abf9d25a2e55614e22b5e819ed4b21ea55cf31aabe6df67')
```
Comment (HadrienG2) [inline var/spack/repos/builtin/packages/hepmc/package.py]: So, err... I get completely different checksums on my machine.

```
hadrien@linux-2ak3:~/Software/spack> curl http://hepmc.web.cern.ch/hepmc/releases/HepMC3-3.1.0.tar.gz | sha256sum
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 20.3M  100 20.3M    0     0  10.2M      0  0:00:01  0:00:01 --:--:-- 10.2M
cd37eed619d58369041018b8627274ad790020a4714b54ac05ad1ebc1a6e7f8a  -
hadrien@linux-2ak3:~/Software/spack> curl http://hepmc.web.cern.ch/hepmc/releases/HepMC3-3.1.1.tar.gz | sha256sum
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 20.6M  100 20.6M    0     0  10.8M      0  0:00:01  0:00:01 --:--:-- 10.8M
2fcbc9964d6f9f7776289d65f9c73033f85c15bf5f0df00c429a6a1d8b8248bb  -
hadrien@linux-2ak3:~/Software/spack> curl http://hepmc.web.cern.ch/hepmc/releases/HepMC3-3.1.2.tar.gz | sha256sum
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 4735k  100 4735k    0     0  7142k      0 --:--:-- --:--:-- --:--:-- 7142k
4133074b3928252877982f3d4b4c6c750bb7a324eb6c7bb2afc6fa256da3ecc7  -
hadrien@linux-2ak3:~/Software/spack> curl http://hepmc.web.cern.ch/hepmc/releases/HepMC3-3.2.0.tar.gz | sha256sum
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 7868k  100 7868k    0     0  10.4M      0 --:--:-- --:--:-- --:--:-- 10.4M
f132387763d170f25a7cc9f0bd586b83373c09acf0c3daa5504063ba460f89fc  -
```

How did you compute those on your side? Should I submit a PR to patch them up, or do you think something genuinely fishy is going on here?

Comment (alalazo) [reply var/spack/repos/builtin/packages/hepmc/package.py]: @HadrienG2 Please submit a PR. I also get your same checksums, apologies for not double checking this (we should definitely automate this check)

@darmac :point_up:

Comment (darmac) [reply var/spack/repos/builtin/packages/hepmc/package.py]: Really strange, I tried again and have a totally different sha256sum with you.
`spack checksum hepmc 3.2.0 3.1.2 3.1.1 3.1.0`
```
==> Checksummed 4 versions of hepmc:

    version('3.2.0', sha256='b59f301e2660db3c81e4fd4aa0701cc010463691ac57424edf4301e16a5944b4')
    version('3.1.2', sha256='7df91b862cead2657ee54ab043ecf0a8f46da354c8a7ea90d8db2c435904bce0')
    version('3.1.1', sha256='46098a7e9717921e436c265836c914d515cb831dde898f3f1d129468df7982e7')
    version('3.1.0', sha256='ac02977d6bde18398abf9d25a2e55614e22b5e819ed4b21ea55cf31aabe6df67')
```

`curl with the same cmd`
```
[root@Estuary-CentOS8 spack]# ls
bin  CHANGELOG.md  COPYRIGHT  etc  lib  LICENSE-APACHE  LICENSE-MIT  NOTICE  opt  README.md  share  var
[root@Estuary-CentOS8 spack]# curl http://hepmc.web.cern.ch/hepmc/releases/HepMC3-3.1.0.tar.gz | sha256sum
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 67.8M    0 67.8M    0     0  1539k      0 --:--:--  0:00:45 --:--:-- 9262k
ac02977d6bde18398abf9d25a2e55614e22b5e819ed4b21ea55cf31aabe6df67  -
```

`wget tar.gz`
```
[root@Estuary-CentOS8 spack]# wget http://hepmc.web.cern.ch/hepmc/releases/HepMC3-3.1.0.tar.gz
--2020-02-13 17:21:01--  http://hepmc.web.cern.ch/hepmc/releases/HepMC3-3.1.0.tar.gz
Resolving proxysg.huawei.com (proxysg.huawei.com)... 172.18.157.51
Connecting to proxysg.huawei.com (proxysg.huawei.com)|172.18.157.51|:8080... connected.
Proxy request sent, awaiting response... 200 OK
Length: unspecified [application/x-gzip]
Saving to: ‘HepMC3-3.1.0.tar.gz’

HepMC3-3.1.0.tar.gz                                    [                                                             <=>                                                 ]  67.87M  10.6MB/s    in 35s

2020-02-13 17:21:43 (1.93 MB/s) - ‘HepMC3-3.1.0.tar.gz’ saved [71168000]

[root@Estuary-CentOS8 spack]# sha256sum HepMC3-3.1.0.tar.gz
ac02977d6bde18398abf9d25a2e55614e22b5e819ed4b21ea55cf31aabe6df67  HepMC3-3.1.0.tar.gz
```

Comment (darmac) [reply var/spack/repos/builtin/packages/hepmc/package.py]: I tried hepmc2.06.09 , also different with the package.py checksum.

```
curl "http://hepmc.web.cern.ch/hepmc/releases/hepmc2.06.09.tgz" | sha256sum
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 11.7M    0 11.7M    0     0   992k      0 --:--:--  0:00:12 --:--:-- 2474k
663f73ca03411dc3b9f81be6e91170bb73147ab92d003f48305bc3c2b4875050  -
```

`spack edit hepmc`
```
version('2.06.09', sha256='e0f8fddd38472c5615210894444686ac5d72df3be682f7d151b562b236d9b422')
```

Comment (darmac) [reply var/spack/repos/builtin/packages/hepmc/package.py]: I changed a network environment and got the same sha256 with you.
```
# curl http://hepmc.web.cern.ch/hepmc/releases/HepMC3-3.1.0.tar.gz | sha256sum
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 20.3M  100 20.3M    0     0   8105      0  0:43:46  0:43:46 --:--:--  3138
cd37eed619d58369041018b8627274ad790020a4714b54ac05ad1ebc1a6e7f8a  -
```

So I'm sorry, I think there is some risk for my network, and I will check all patch I commited in these days with another network environment.

Comment (alalazo) [reply var/spack/repos/builtin/packages/hepmc/package.py]: @darmac I just double checked most of your PRs from last week and the other ones seems to have correct hashes.

Comment (darmac) [reply var/spack/repos/builtin/packages/hepmc/package.py]: That's nice of you, thanks.

**Pull request merged**